CanvasPlusPlayground/Views/CourseAnnouncementDetailView.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  CourseAnnouncementDetailView.swift
+//  CanvasPlusPlayground
+//
+//  Created by Ethan Fox on 9/13/24.
+//
+
+import SwiftUI
+import UIKit
+
+struct CourseAnnouncementDetailView: View {
+    let announcement: Announcement
+    var body: some View {
+        ScrollView {
+            VStack {
+                Text(announcement.title ?? "NULL_TITLE")
+                    .font(.title)
+                Text(announcement.createdAt?.formatted() ?? "NULL_DATE")
+                    .font(.subheadline)
+                AttributedText(.html(withBody: announcement.message ?? "NULL_MESSAGE"))
+                
+            }
+        }
+        .padding()
+    }
+}
+
+
+// All the code below is used to format HTML into a Swift View. I did not write it
+// Here is my source: https://medium.com/@thomsmed/rendering-html-in-swiftui-65e883a63571
+
+struct AttributedText: UIViewRepresentable {
```

Comment:
If you decide to keep this you can move it all into another file, maybe in a "Helpers" group/folder.